### DISCUSSION TITLE
[SOLVED] convert PSD file with wrong result

### DISCUSSION THREAD

**Discussion by noxxxxxxxx** [ImageMagick/ImageMagick#5227] [Help] [answered]
imagemagick version `ImageMagick 7.1.0-26 Q16-HDRI x86_64 2022-02-21`

convert `demo1.psd` file to png get wrong thumbnail, it should be 
<img width="88" alt="image" src="https://user-images.githubusercontent.com/37919656/173771150-e664e3a8-4c13-4c39-87de-0a7417a113c7.png">

but I got this with the command `convert demo1.psd -flatten thumbnail.png`

<img width="299" alt="image" src="https://user-images.githubusercontent.com/37919656/173771293-27b444d2-c785-41cf-8bcc-5b805d3b5982.png">

also got the error msg when I try to convert   `demo2.psd` 

`Command failed: identify: improper image header /convert/444a7dc023f385043f55cb35d19e7ad0.psd' @ error/psd.c/ReadPSDLayersInternal/1984`

Attach file: [https://www.dropbox.com/s/oc0eqngudpmrn2t/Archive.zip?dl=0](dropbox)

**Comment by fmw42** [comment]
Add [0] to the input.

`convert demo1.psd[0] ...  thumbnail.png`

... means any processing commands you want to add

**Reply by noxxxxxxxx** [reply]
I tried but get error msg:

`zsh: no matches found: demo1.psd[0]`

**Comment by fmw42** [comment]
What was your exact command?

**Reply by noxxxxxxxx** [reply]
`convert demo1.psd[0]  thumbnail.png`

I tried `convert demo1.psd  thumbnail.png` will generate 367 files , from `thumbnail-0.png` to `thumbnail-367.png`,  `thumbnail-0.png` seems the right thumbnail image.

<img src="https://user-images.githubusercontent.com/37919656/174006774-f113a707-8c70-4f6e-a9a4-6535c1dd41a5.png" width="200px">

**Comment by snibgo** [comment, marked as answer]
Perhaps zsh is interpreting [ and ] as a pattern for filename generation. See https://zsh.sourceforge.io/Doc/Release/Expansion.html#Filename-Generation Try quoting the filename, including [0]. I don't know if they should be double or single quote characters.
```
convert "demo1.psd[0]" thumbnail.png
```
But as you use IM v7, I suggest using "magick" instead of "convert":
```
magick "demo1.psd[0]" thumbnail.png
```

**Reply by noxxxxxxxx** [reply]
Wow, thanks a lot. with double quote `magick "demo1.psd[0]" thumbnail.png`. It works!
